File: src/main/scala/jk_5/nailed/server/scoreboard/NailedScore.scala
/*
 * Nailed, a Minecraft PvP server framework
 * Copyright (C) jk-5 <http://github.com/jk-5/>
 * Copyright (C) Nailed team and contributors <http://github.com/nailed/>
 *
 * This program is free software: you can redistribute it and/or modify it
 * under the terms of the MIT License.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
 * for more details.
 *
 * You should have received a copy of the MIT License along with
 * this program. If not, see <http://opensource.org/licenses/MIT/>.
 */

package jk_5.nailed.server.scoreboard

import jk_5.nailed.api.scoreboard.Score
import jk_5.nailed.api.util.Checks
import net.minecraft.network.play.server.S3CPacketUpdateScore

/**
 * No description given
 *
 * @author jk-5
 */
class NailedScore(val owner: NailedObjective, val name: String) extends Score {
  Checks.notNull(owner, "owner may not be null")
  Checks.notNull(name, "name may not be null")
  Checks.check(name.length() <= 16, "name may not be longer than 16")

  var value: Int = 0
  override def setValue(value: Int){
    this.value = value
    update()
  }
  override def addValue(value: Int) = this.setValue(this.value + value)

  override def update(){
    val p = new S3CPacketUpdateScore
    p.field_149329_a = this.name
    p.field_149327_b = this.owner.id
    p.field_149328_c = this.value
    p.field_149326_d = 0
    owner.manager.sendPacket(p)
  }
}

File: src/main/java/jk_5/nailed/server/scoreboard/NailedScore.java
package jk_5.nailed.server.scoreboard;

import jk_5.nailed.api.scoreboard.Score;
import jk_5.nailed.api.util.Checks;
import net.minecraft.network.play.server.S3CPacketUpdateScore;

import javax.annotation.Nonnull;

public class NailedScore implements Score {

    private final NailedObjective owner;
    private final String name;

    private int value;

    public NailedScore(NailedObjective owner, String name) {
        this.owner = owner;
        this.name = name;

        Checks.notNull(owner, "owner may not be null");
        Checks.notNull(name, "name may not be null");
        Checks.check(name.length() <= 16, "name may not be longer than 16");
    }

    @Nonnull
    @Override
    public String getName() {
        return this.name;
    }

    @Override
    public int getValue() {
        return this.value;
    }

    @Override
    public void setValue(int value) {
        this.value = value;
        this.update();
    }

    @Override
    public void addValue(int value) {
        this.setValue(this.value + value);
    }

    @Override
    public void update() {
        S3CPacketUpdateScore p = new S3CPacketUpdateScore();
        p.name = this.name;
        p.objective = this.owner.id();
        p.value = this.value;
        p.action = S3CPacketUpdateScore.Action.CHANGE;
        owner.manager().sendPacket(p);
    }
}
